odl/test/set/domain_test.py

```diff
@@ -179,7 +179,22 @@ def test_element():
     assert set_.element() in set_
 
 
+def _test_eq(x, y):
+    """Test equality of x and y."""
+    assert x == y
+    assert not x != y
+    assert hash(x) == hash(y)
+
+
+def _test_neq(x, y):
+    """Test non-equality of x and y."""
+    assert x != y
+    assert not x == y
+    assert hash(x) != hash(y)
```

Comment:
This is, strictly speaking, not a requirement. But we didn't get any issues in the tests and it should hold "in general", so I'll keep it.